app/reducers/surveys.ts

```diff
@@ -18,126 +20,113 @@ import type {
-> & {
-  questions: (Omit<DetailedSurvey['questions'][number], 'id'> & {
-    options: Omit<DetailedSurvey['questions'][number]['options'], 'id'>;
-  })[];
-  templateType: EventType;
-};
-
-export const selectSurveyTemplate = createSelector(
-  (state: RootState) => selectSurveys(state),
-  (_: RootState, props: SurveyTemplateProps) => props.templateType,
-  (surveys, templateType) => {
-    const template = surveys.find(
-      (survey) => survey.templateType === templateType,
-    );
-    if (!template) return undefined;
+export type TransformedSurveyTemplate = Overwrite<
+  Omit<DetailedSurvey, 'id' | 'event' | 'activeFrom'>,
+  {
+    questions: Overwrite<
+      Omit<SurveyQuestion, 'id'>,
+      {
+        options: Omit<SurveyQuestionOption, 'id'>[];
+      }
+    >[];
+  }
+>;
+export const selectSurveyTemplateByType = createSelector(
+  selectSurveysByField('templateType').single,
+  (template) => {
+    if (!template || !('actionGrant' in template)) return undefined;
     const questions = (template.questions || []).map((question) => ({
       ...omit(question, 'id'),
       options: question.options.map((option) => omit(option, 'id')),
     }));
     return {
       ...omit(template, ['id', 'event', 'activeFrom']),
       questions,
-    } as SurveyTemplate;
+    } satisfies TransformedSurveyTemplate;
   },
 );
 
 export const useFetchedTemplate = (
   prepareId: string,
   templateType?: EventType,
-): SurveyTemplate | undefined => {
+): TransformedSurveyTemplate | undefined => {
   const dispatch = useAppDispatch();
   usePreparedEffect(
     `useFetchedTemplate-${prepareId}`,
     () => templateType && dispatch(fetchTemplate(templateType)),
     [templateType],
   );
   return useAppSelector((state: RootState) =>
-    templateType ? selectSurveyTemplate(state, { templateType }) : undefined,
+    templateType ? selectSurveyTemplateByType(state, templateType) : undefined,
   );
 };
 
+type FetchSurveyResult<ST = UnknownSurvey> = {
+  survey: ST | undefined;
+  event: EventForSurvey | undefined;
+};
 export function useFetchedSurvey(
   prepareId: string,
-  surveyId?: EntityId,
-  token?: string,
-): SelectedPublicResultsSurvey | undefined;
+  surveyId: EntityId,
+  token: string,
+): FetchSurveyResult<PublicResultsSurvey>;
 export function useFetchedSurvey(
   prepareId: string,
-  surveyId?: EntityId,
-): SelectedSurvey | undefined;
+  surveyId: EntityId,
+): FetchSurveyResult<DetailedSurvey>;
 export function useFetchedSurvey(
   prepareId: string,
-  surveyId?: EntityId,
+  surveyId: EntityId,
   token?: string,
 ) {
   const dispatch = useAppDispatch();
   usePreparedEffect(
     `useFetchedSurvey-${prepareId}`,
     () =>
-      surveyId
-        ? token
-          ? dispatch(fetchWithToken(surveyId, token))
-          : dispatch(fetchSurvey(surveyId))
-        : Promise.resolve(),
+      token
+        ? dispatch(fetchWithToken(surveyId, token))
+        : dispatch(fetchSurvey(surveyId))[surveyId],
```

Comment:
```suggestion
        : dispatch(fetchSurvey(surveyId)),
```
Am I missing something here?

Reply:
hmm, that's weird